File: C++版/049-字母异位词分组.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> result;   //结果
        unordered_map<string,vector<string>> rec;  //排序后的词作为键值
        for(int i=0;i<strs.size();i++)
        {
            string str = strs[i];
            sort(str.begin(),str.end());  //对单词按字典序排序
            rec[str].push_back(strs[i]);
        }
        for(auto i:rec)   //auto是智能指针
        {
            result.push_back(i.second);
        }
        return result;
    }
};

File: Java/049-字母异位词分组.java
import java.util.*;

class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> result = new ArrayList<>(); // 结果
        HashMap<String, List<String>> rec = new HashMap<>(); // 排序后的词作为键
        for (int i = 0; i < strs.length; i++) {
            String str = strs[i];
            byte[] a = str.getBytes();
            Arrays.sort(a); // 对单词按字典序排序
            str = new String(a);
            List<String> t = rec.get(str);
            if (t == null)
                t = new ArrayList<>();
            t.add(strs[i]);
            rec.put(str, t);
        }
        Iterator iter = rec.entrySet().iterator();
        while (iter.hasNext()) // auto是智能指针
        {
            Map.Entry<String, List<String>> entry = (Map.Entry) iter.next();
            result.add(entry.getValue());
        }
        return result;
    }
}